comp/api/api/apiimpl/internal/config/endpoint.go

```diff
@@ -38,12 +49,32 @@ type configEndpoint struct {
 
 func (c *configEndpoint) getConfigValueHandler(w http.ResponseWriter, r *http.Request) {
 	vars := gorilla.Vars(r)
-	path := vars["path"]
 	// escape in case it contains html special characters that would be unsafe to include as is in a response
 	// all valid config paths won't contain such characters so for a valid request this is a no-op
-	path = html.EscapeString(path)
+	path := html.EscapeString(vars["path"])
+
+	authorized := false
+	if _, ok := c.authorizedConfigPaths[path]; ok {
+		authorized = true
+	} else {
+		// check to see if the requested path matches any of the authorized paths
+		//
+		// if an authorized path ends in a period, it authorizes any paths under that prefix,
+		// including the prefix itself: an authorized path of `my.config` allows for reading
+		// `my.config`, `my.config.field1`, `my.config.field2`, etc
+		for authorizedPath := range c.authorizedConfigPaths {
+			isPrefixPathRule := isPrefixPath(authorizedPath)
+			isPrefixSubpath := strings.HasPrefix(path, authorizedPath)
+			isPrefixPathRoot := pathToPrefixPath(path) == authorizedPath
+
+			if isPrefixPathRule && (isPrefixSubpath || isPrefixPathRoot) {
+				authorized = true
+				break
+			}
+		}
```

Comment:
This logic could be simplified if we consider that any subconfig of an authorized config can be queried, no need for the `.` suffix in the authorized keys
What do you think ?

The condition in the `for` loop becomes just `if strings.HasPrefix(path, authorizedPath + prefixPathSuffix)`

Reply:
We still have to check `path + prefixPathSuffix` against `authorizedPath`, otherwise we can't query maps that are authorized via a prefix rule.

(If authorized rule is `ha.`, then only querying `ha.` would work with your idea... but the config will not return anything if we do `Get("ha.")`, since it has to be `"ha"`)

Reply:
I'm not sure I follow, my proposal is to have `ha` in the authorized map directly, so that if you query `"ha"` it works, and if you query `"ha.something"` it works too because `"ha.something"` has prefix `"ha" + prefixPathSuffix`

Reply:
Ah, I see what you mean now.

Yeah, this would be cleaner. 👍🏻

Reply:
Addressed in [937c477](https://github.com/DataDog/datadog-agent/pull/22290/commits/937c47796c23b71a5b21255711b949d0482bcd13).